apps/api/src/campaign/campaign.service.ts

```diff
@@ -57,17 +57,29 @@ export class CampaignService {
   }
 
   async listAllCampaigns(): Promise<Campaign[]> {
-    return await this.prisma.campaign.findMany({
+    const campaigns = await this.prisma.campaign.findMany({
       orderBy: {
         endDate: 'asc',
       },
       include: {
         campaignType: { select: { name: true } },
         beneficiary: { select: { person: { select: { firstName: true, lastName: true } } } },
         coordinator: { select: { person: { select: { firstName: true, lastName: true } } } },
-        vaults: { select: { id: true, amount: true } },
+        vaults: { select: { donations: { select: { amount: true } }, amount: true } },
+        incomingTransfers: { select: { amount: true } },
+        outgoingTransfers: { select: { amount: true } },
       },
     })
+    for (const campaign of campaigns) {
+      let campaignAmountReached = 0
+      for (const vault of campaign.vaults) {
+        for (const donation of vault.donations) {
+          campaignAmountReached += donation.amount
+        }
+      }
+      campaign['reachedAmount'] = campaignAmountReached
```

Comment:
The frontend expects that `reachedAmount` is part of `campaign.summary.reachedAmount`

Do we need to update it there as well?